File: python/tests/fakes.py
class FakeTransport:
    def __init__(self, responses):
        self._responses = list(responses)
        self.requests = []

    def request(self, method, path, query=None, json_body=None):
        self.requests.append(
            {"method": method, "path": path, "query": query, "json_body": json_body}
        )
        return self._responses.pop(0)

File: node/tests/fakes.ts
import type { HttpResponse, Transport } from '../src/http';

export class FakeTransport implements Transport {
  public requests: Array<{
    method: string;
    path: string;
    query?: Record<string, string | number>;
    jsonBody?: unknown;
  }> = [];

  constructor(private responses: HttpResponse[]) {}

  async request(
    method: string,
    path: string,
    query?: Record<string, string | number>,
    jsonBody?: unknown
  ): Promise<HttpResponse> {
    this.requests.push({ method, path, query, jsonBody });
    const response = this.responses.shift();
    if (!response) {
      throw new Error('FakeTransport: no more queued responses');
    }
    return response;
  }
}
